src/platform.ts

```diff
@@ -469,6 +475,17 @@ export function getWindowsSystemPowerShellPath(systemFolderName: string) {
         "powershell.exe");
 }
 
+function fileExistsSync(filePath: string): boolean {
```

Comment:
The node docs aren't wrong about this kinda being an anti-pattern.

> fs.exists() is an anachronism and exists only for historical reasons. There should almost never be a reason to use it in your own code.
> 
> In particular, checking if a file exists before opening it is an anti-pattern that leaves you vulnerable to race conditions: another process may remove the file between the calls to fs.exists() and fs.open(). Just open the file and handle the error when it's not there.
> 
> fs.exists() will be deprecated.

We're essentially making our own `fs.existsSync` and I question if we need it. As the code is written right now, this does seem to be the best option, but I'd want to review why we're checking if a file exists, then checking if a folder exists, etc.